mm2src/coins/eth/eth_swap_v2/eth_maker_swap_v2.rs

```diff
@@ -0,0 +1,490 @@
+            )
+            .await?;
+
+        let tx_from_rpc = self
+            .transaction(TransactionId::Hash(args.maker_payment_tx.tx_hash()))
+            .await?;
+        let tx_from_rpc = tx_from_rpc.as_ref().ok_or_else(|| {
+            ValidatePaymentError::TxDoesNotExist(format!(
+                "Didn't find provided tx {:?} on ETH node",
+                args.maker_payment_tx.tx_hash()
+            ))
+        })?;
+        validate_from_to_and_status(
+            tx_from_rpc,
+            maker_address,
+            maker_swap_v2_contract,
+            maker_status,
+            MakerPaymentStateV2::PaymentSent as u8,
+        )?;
+
+        let validation_args = {
+            let amount = wei_from_big_decimal(&args.amount, self.decimals)?;
+            MakerValidationArgs {
+                swap_id,
+                amount,
+                taker: self.my_addr().await,
+                taker_secret_hash: args.taker_secret_hash,
+                maker_secret_hash: args.maker_secret_hash,
+                payment_time_lock: args.time_lock,
+            }
+        };
+        match self.coin_type {
+            EthCoinType::Eth => {
+                let function = MAKER_SWAP_V2.function(ETH_MAKER_PAYMENT)?;
+                let decoded = decode_contract_call(function, &tx_from_rpc.input.0)?;
+                validate_eth_maker_payment_data(&decoded, &validation_args, function, tx_from_rpc.value)?;
+            },
+            EthCoinType::Erc20 { token_addr, .. } => {
+                let function = MAKER_SWAP_V2.function(ERC20_MAKER_PAYMENT)?;
+                let decoded = decode_contract_call(function, &tx_from_rpc.input.0)?;
+                validate_erc20_maker_payment_data(&decoded, &validation_args, function, token_addr)?;
+            },
+            EthCoinType::Nft { .. } => {
+                return MmError::err(ValidatePaymentError::ProtocolNotSupported(
+                    "NFT protocol is not supported for ETH and ERC20 Swaps".to_string(),
+                ));
+            },
+        }
+        Ok(())
+    }
+
+    pub(crate) async fn refund_maker_payment_v2_timelock_impl(
+        &self,
+        args: RefundMakerPaymentTimelockArgs<'_>,
+    ) -> Result<SignedEthTx, TransactionErr> {
+        let token_address = self
+            .get_token_address()
+            .map_err(|e| TransactionErr::Plain(ERRL!("{}", e)))?;
+        let gas_limit = self
+            .gas_limit_v2
+            .gas_limit(
+                &self.coin_type,
+                EthPaymentType::MakerPayments,
+                PaymentMethod::RefundTimelock,
+            )
+            .map_err(|e| TransactionErr::Plain(ERRL!("{}", e)))?;
+
+        let maker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.maker_swap_v2_contract)
+            .ok_or_else(|| TransactionErr::Plain(ERRL!("Expected swap_v2_contracts to be Some, but found None")))?;
```

Comment:
```suggestion
     let maker_swap_v2_contract = self
    .swap_v2_contracts
    .as_ref()
    .ok_or_else(|| TransactionErr::Plain(ERRL!("Expected swap_v2_contracts to be Some, but found None")))?
    .maker_swap_v2_contract;
```

Reply:
Thanks for catching, also as_ref isnt necessary 
fixed here and in other places https://github.com/KomodoPlatform/komodo-defi-framework/pull/2211/commits/57e75ed9354341cff841c4391871b75899f420b8